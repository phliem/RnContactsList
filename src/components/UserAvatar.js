// @flow
import React from 'react';
import { Image, StyleSheet } from 'react-native';

// Constants & Services
import AVATARS from '../res/avatars';
import COLORS from '../res/colors';
import SIZES from '../res/sizes';

type PROPS = {
  id: string,
  isActive: boolean,
};

const UserAvatar = (props: PROPS) => {
  const imageStyle = [styles.img];
  if (props.isActive) {
    imageStyle.push(styles.imgActive);
  }

  return <Image style={imageStyle} source={AVATARS[props.id]} />;
};

const styles = StyleSheet.create({
  img: {
    width: SIZES.avatarWidth,
    height: SIZES.avatarHeight,
  },
  imgActive: {
    borderWidth: 3,
    borderColor: COLORS.highlight,
    borderRadius: SIZES.avatarWidth / 2,
  },
});

export default UserAvatar;
