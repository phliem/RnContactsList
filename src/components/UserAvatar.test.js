import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';

// Components
import UserAvatar from './UserAvatar';

describe('Should test UserAvatar Component', () => {
  test('render and match snapshot', () => {
    const testRender = renderer.create(
      <UserAvatar id={'AVATAR_ID'} isActive={false} />,
    );
    const testInstance = testRender.toJSON();
    expect(testInstance).toMatchSnapshot();
  });
});
