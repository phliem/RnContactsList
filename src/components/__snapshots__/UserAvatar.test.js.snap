// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test UserAvatar Component render and match snapshot 1`] = `
<Image
  style={
    Array [
      Object {
        "height": 64,
        "width": 64,
      },
    ]
  }
/>
`;
